### PR TITLE
Fix sequencer number bug

### PR DESCRIPTION
**Pull request by asonnino**
* Fix bug https://github.com/MystenLabs/sui/issues/1600
* Make a unit test to ensure the bug is fixed

### CODE REVIEW

File: sui_core/src/authority/authority_store.rs [hunk 1]
```diff
@@ -834,10 +834,10 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
         let (sequenced_to_write, schedule_to_write): (Vec<_>, Vec<_>) = ids
             .zip(versions.iter())
             .map(|(id, v)| {
-                let version = v.unwrap_or_else(SequenceNumber::new);
+                let version = v.unwrap_or_else(|| SequenceNumber::from(1));
```
Comment (lxfind) [inline sui_core/src/authority/authority_store.rs:837]: There is `OBJECT_START_VERSION` in object.rs, may be good to use that constant
Could you add comments here on why this must be 1?

Comment (asonnino) [reply sui_core/src/authority/authority_store.rs:837]: fixed

File: sui_core/src/authority/authority_store.rs [hunk 2]
```diff
@@ -834,10 +834,10 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
         let (sequenced_to_write, schedule_to_write): (Vec<_>, Vec<_>) = ids
             .zip(versions.iter())
             .map(|(id, v)| {
-                let version = v.unwrap_or_else(SequenceNumber::new);
+                let version = v.unwrap_or_else(|| OBJECT_START_VERSION);
```
Comment (lxfind) [inline sui_core/src/authority/authority_store.rs:837]: Could you add comment here on why it must be OBJECT_START_VERSION?

Comment (asonnino) [reply sui_core/src/authority/authority_store.rs:840]: Fixed

File: test_utils/src/messages.rs [hunk 3]
```diff
@@ -59,19 +64,68 @@ pub fn test_shared_object_transactions() -> Vec<Transaction> {
...
     transactions
 }
 
-/// Make a test certificates for each shared-object transaction.
-pub async fn test_shared_object_certificates() -> Vec<CertifiedTransaction> {
+/// Make a transaction to publish a test move contracts package.
+pub fn publish_move_package_transaction(gas_object: Object) -> Transaction {
+    let build_config = BuildConfig::default();
+    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    path.push("../sui_programmability/examples/basics");
```
Comment (lxfind) [inline test_utils/src/messages.rs:79]: Might want to make this an argument to make this function more generic.

Comment (asonnino) [reply test_utils/src/messages.rs:79]: Good point https://github.com/MystenLabs/sui/issues/974

**Pull request merged**